src/spyglass/spikesorting/v0/spikesorting_curation.py

```diff
@@ -604,11 +604,12 @@ def _compute_metric(self, waveform_extractor, metric_name, **metric_params):
 
         is_nn_iso = metric_name == "nn_isolation"
         is_nn_overlap = metric_name == "nn_noise_overlap"
+        is_num_spikes = metric_name == "num_spikes"
         min_spikes = metric_params.get("min_spikes", 10)
 
         for unit_id in waveform_extractor.sorting.get_unit_ids():
             # checks to avoid bug in spikeinterface 0.98.2
-            if num_spikes[unit_id] < min_spikes:
+            if num_spikes[unit_id] < min_spikes and not is_num_spikes:
```

Comment:
I may be wrong about this since it's been a while since I wrote this (please correct me if that is the case). It seems we need special handling of this case (i.e. number of spikes for a unit is below `min_spikes`) only for computing `nn_isolation` and `nn_overlap` metrics. So instead of adding `not is_num_spikes`, how about changing to code to check for only those two conditions and have all the others go to ? That way we won't have to fix this again when we compute another metric (like `snr`, `isi_violation`, `peak_offset` etc)